lib/resty/upstream/healthcheck.lua

```diff
@@ -513,10 +512,8 @@ local function preprocess_peers(peers)
         local p = peers[i]
         local name = p.name
         if name then
-            local idx = str_find(name, ":", 1, true)
-            if idx then
-                p.host = sub(name, 1, idx - 1)
-                p.port = tonumber(sub(name, idx + 1))
+            if name:sub(-1, -1)~=']' then
```

Comment:
This involves a table field lookup. Better use the form `func(self, ...)` instead.

Also, creating a substring of a single character is expensive. Better use `string.char` here instead.